src/emacs/lean-info.el

```diff
@@ -67,6 +67,26 @@
       (lambda (&key record)
         (funcall cont record))))))
 
+(defun lean-get-info-record-at-point-sync ()
+  "Get info-record at point synchronously."
+  (with-demoted-errors "lean get info: %s"
+    (lean-server-send-synchronous-command
+     'info (list :file_name (buffer-file-name)
+                 :line (line-number-at-pos)
+                 :column (lean-line-offset)))))
+
+(defun lean-info-right-click-find-definition ()
+  "Offer to jump to definition of right-click target."
+  (interactive)
+  (let* ((info-record (plist-get (lean-get-info-record-at-point-sync) :record))
```

Comment:
I agree that regexen are a bad solution here, and am working on the timeout version.

The issue with always showing all things in the right-click menu is that many of them are being dynamically computed by Lean metaprograms right now, such as the hole commands. So this means that a solution for asking the server is necessary.

Emacs should do better than VS Code where possible :-) The whole point of a contextual menu is that it gives the commands that are relevant to some particular context. After all, what's the good of knowing that a "Find definition" command exists that will throw an error?